pkg/cache/components/scope_informer_test.go

```diff
@@ -0,0 +1,272 @@
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("ScopeInformer Unit Tests", func() {
+	// create the informer factory
+	When("Creating a new ScopeInformer", func() {
+		var (
+			si              *ScopeInformer
+			ctx             context.Context
+			cancel          context.CancelFunc
+			informerFactory informers.SharedInformerFactory
+		)
+
+		BeforeEach(func() {
+			informerFactory = informers.NewSharedInformerFactory(k8sClient, 10*time.Second)
+			Expect(informerFactory).NotTo(BeNil())
+			// Create a new informers.GenericInformer
+			inf, err := informerFactory.ForResource(corev1.SchemeGroupVersion.WithResource("pods"))
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(inf).ShouldNot(BeNil())
+
+			// Expected context
+			ctx, cancel = context.WithCancel(context.Background())
+
+			// Create the new ScopeInformer
+			si = NewScopeInformer(inf)
+			Expect(si.informer).Should(Equal(inf))
+			Expect(si.ctx).Should(BeEquivalentTo(ctx))
+			Expect(si.cancel).ShouldNot(BeNil())
+			Expect(si.dependents).ShouldNot(BeNil())
+			Expect(si.eventListeners).ShouldNot(BeNil())
+		})
+
+		AfterEach(func() {
+			cancel()
+			si.Terminate()
+		})
+
+		When("Running the ScopeInformer", func() {
+			It("Should start the ScopeInformer.informer", func() {
+				// start running the ScopeInformer
+				go si.Run()
+
+				started := func() bool {
+					return si.HasSynced()
+				}
+
+				Eventually(started).Should(BeTrue())
+			})
+
+			When("Terminating the ScopeInformer", func() {
+				It("Should no longer be running", func() {
+					si.Terminate()
+					terminated := func() bool {
+						return errors.Is(si.ctx.Err(), context.Canceled)
+					}
+
+					Eventually(terminated).Should(BeTrue())
+				})
+			})
+		})
+
+		When("Adding a Dependent", func() {
+			var dependent v1.Object
+			BeforeEach(func() {
+				dependent = &corev1.Namespace{ObjectMeta: v1.ObjectMeta{UID: types.UID("test-dep"), Name: "test-ns"}}
+				si.AddDependent(dependent)
+			})
+
+			It("Should be in the map returned by ScopeInformer.GetDependents()", func() {
+				siDeps := si.GetDependents()
+				Expect(siDeps).Should(HaveLen(1))
+				Expect(siDeps).Should(HaveKeyWithValue(dependent.GetUID(), dependent))
+			})
+
+			It("Should have the dependent", func() {
+				Expect(si.HasDependent(dependent)).Should(BeTrue())
+			})
+
+			When("Removing the Dependent", func() {
+				BeforeEach(func() {
+					si.RemoveDependent(dependent)
+				})
+
+				It("Should not have the dependent", func() {
+					Expect(si.HasDependent(dependent)).Should(BeFalse())
+				})
+			})
+		})
+
+		When("Calling AddEventHandler()", func() {
+			It("Should be added to the ScopeInformer.informer if it hasn't been before", func() {
```

Comment:
What should be added? Maybe `It("Should add THE THING to the ScopeInformer...` where THE THING is named for what it is we're adding :)

Reply:
Good point! I think I did this in later tests...

Should be fixed as of https://github.com/everettraven/telescopia/pull/14/commits/da1f64c076a3c3f721be56e3ac779ca229808298